src/main/scala/markatta/futiles/CancellableFuture.scala

```diff
@@ -0,0 +1,81 @@
+package markatta.futiles
+
+import scala.concurrent._
+import scala.util.Try
+import java.util.concurrent.{Callable, FutureTask}
+import scala.concurrent.duration.Duration
+
+trait CancellableFuture[T] extends Future[T] {
+
+  /** Attempts to cancel the underlying [[scala.concurrent.Future]]. Note that this is a best effort attempt
+    */
+  def cancel(): Unit
+}
+
+private[futiles] class CancellableFutureImpl[T](executionContext: ExecutionContext, block: => T)
+    extends CancellableFuture[T] {
+  private val promise = Promise[T]()
+
+  def delegate: Future[T] = promise.future
```

Comment:
Not sure if you want this to be private or public? Typically from what I have seen delegates tend to be exposed but I don't have strong thoughts either way.